### PR TITLE
Fixed scheduler for PyTorch

### PR DESCRIPTION
**Pull request by ViktoriiaRomanova**
This is one of the versions of how to fix scheduler. 
In this version for scheduler have no difference between epoch and round (e.g. if number_epoch_per_round = 2 and round = 3 => scheduler count it as 6 steps)

### CODE REVIEW

File: openfl-tutorials/interactive_api/PyTorch_DogsCats_ViT/workspace/PyTorch_DogsCats_ViT.ipynb [hunk 1]
```diff
@@ -374,10 +374,15 @@
...
+    "from torch.optim.lr_scheduler import StepLR\n",
+    "\n",
+    "\n",
     "# loss function\n",
     "criterion = nn.CrossEntropyLoss()\n",
     "# optimizer\n",
-    "optimizer = optim.Adam(model.parameters(), lr=lr)"
+    "optimizer = optim.Adam(model.parameters(), lr=lr)\n",
+    "\n",
+    "scheduler = StepLR(optimizer, step_size=1, gamma=0.1)"
```
Comment (dmitryagapov) [inline openfl-tutorials/interactive_api/PyTorch_DogsCats_ViT/workspace/PyTorch_DogsCats_ViT.ipynb]: Maybe we can add `verbose=True` to show that it works.

**Review by dmitryagapov — APPROVED** [review]

File: openfl/plugins/frameworks_adapters/pytorch_adapter.py [hunk 2]
```diff
@@ -70,14 +70,13 @@ def _set_optimizer_state(optimizer, device, derived_opt_state_dict):
...
-    # Setting default values.
-    # All optimizer.defaults are considered as not changing over course of
-    # training.
-    for group in temp_state_dict['param_groups']:
-        for k, v in optimizer.defaults.items():
-            group[k] = v
+    # Setting other items from the param_groups
+    # getting them from the local optimizer
+    # (expand_derived_opt_state_dict sets only 'params')
+    for i, group in enumerate(optimizer.param_groups):
```
Comment (aleksandr-mokrov) [inline openfl/plugins/frameworks_adapters/pytorch_adapter.py:76]: You can replace two for by one: `for idx, (k, v) in enumerate(optimizer.param_groups.items()):`.

**Comment by ViktoriiaRomanova** [comment]
Jenkins please retry a build

**Review by aleksandr-mokrov — APPROVED** [review]

**Comment by igor-davidyuk** [comment]
Will this method also work with TensorFlow and Keras?

**Comment by ViktoriiaRomanova** [comment]
No, this fix is only for PyTorch

**Comment by igor-davidyuk** [comment]
> No, this fix is only for PyTorch

Could you please investigate if it works for a tensorflow example in our repo, as the interface is unified in OpenFL

**Comment by alexey-gruzdev** [comment]
Close due to #305
